common/identity/entryBlockSync.go

```diff
@@ -136,14 +136,24 @@ func (e *EntryBlockSync) UnmarshalBinaryData(p []byte) (newData []byte, err erro
 		return
 	}
 
-	l, err := buf.PopInt()
+	// blockLimit is the mazimum number of Entry Blocks that could fit in the
+	// buffer. Smallest possible Entry Block is 140 bytes.
+	blockLimit := buf.Len() / 140
```

Comment:
Swap ordering of limit and count assignments (?again, unsure if this one would actually be affected, or if the division already covers it enough. pointing out just in case)